common/src/main/java/org/openmrs/analytics/HttpUtil.java

```diff
@@ -49,7 +51,34 @@ public String executeRequest(HttpUriRequest request) {
 		}
 		catch (IOException e) {
 			log.error("Error in opening url: " + request.getURI().toString() + " exception: " + e);
+			httpClient.close();
```

Comment:
Thanks for catching this but should we move this to a `finally` segment? Or maybe do the `try` with the resource to auto-close it.

Reply:
sure